modules/multiprocessExample/main.py

```diff
@@ -0,0 +1,160 @@
+import multiprocessing as mp
+# import pathos.multiprocessing as pm
+import time
+import counter
+
+
+# If the queue is empty then the exit was not requested
+def exitRequested(requestQueue):
+    return not requestQueue.empty()
+
+def counterWorker(counterClass, pause, exitRequest, pipelineOut):
+    """
+    This function is used for multiprocessing.
+    The controller of the process(es) will be referred to as the manager (just to make it easier).
+
+    Parameters
+    ----------
+    counterClass: The class object
+    pause: A mutex lock used to pause/resume this process
+    exitRequest: Queue used for indicating that this process should exit
+    pipelineOut: Queue used for producer data output
+        If this was also a consumer we would have an input queue
+
+    Returns
+    -------
+
+    """
+    print("Counter start!")
+    while (True):
+
+        # If the manager acquires the lock instead,
+        # processes will end up blocking here
+        # This is a turnstile pattern
+        pause.acquire()
+        pause.release()
+
+        # If the class were also a consumer, this code would exist
+        # Queues are process-safe, so no need to protect it
+        # IMPORTANT: If the queue is empty it will be stuck here forever,
+        # This is bad if an exit is requested, so a way to get it unstuck
+        # is to have the manager manually push a few items (see manager below)
+
+        # inData = pipelineIn.get()
+
+        # Do the work
+        # Do not unpack the data here, unpack it in the class method
+        # All work should be done in the class method!
+        # This wrapper function should only be doing process things!
+        # Single Responsibility Principle
+        ret, outData = counterClass.countUp()  # .countUp(inData) if it's also a consumer
+
+        # Something went wrong so we'll skip
+        if (not ret):
+            continue
+
+        print("Inserting data: " + str(outData))
+        # Queues are process-safe, so no need to protect it
+        # IMPORTANT: If the queue is full it will be stuck here forever
+        # This is bad if an exit is requested, so a way to get it unstuck
+        # is to have the manager manually pop a few items (see manager below)
+        pipelineOut.put(outData)
+
+        # Check whether a reset was called
+        # It is shared, which means it's a critical section,
+        # and so this is technically bad because it's not protected by a lock
+        # HOWEVER: Sharing memory between processes is a pain,
+        # and the worst case is that we end up
+        # processing an extra item before exiting and dying
```

Comment:
Explain further that writing a bad value is far worse, but since queue implements its own internal locks it's fine-ish.